R/s3-soma-adat.R

```diff
@@ -57,23 +57,22 @@ NULL
 `[.soma_adat` <- function(x, i, j, drop = TRUE, ...) {
 
   if ( missing(j) ) {
-    # not sub-setting columns
-    return(NextMethod())
-  } else if ( is_intact_attr(x) ) {
-    # sub-setting columns & attributes to worry about
-    if ( length(j) == 1L && j > 0 ) {
-      # if extracting a single column
-      # this behavior may change one day to match `tibbles`
-      # where you output is what you input, i.e. `drop = FALSE` by default
-      return(NextMethod(drop = drop))
-    } else {
-      atts <- attributes(x)
-    }
-  } else {
-    # if attributes already broken
-    return(NextMethod())
+    # if sub-setting rows; nothing special to do
+    .data <- NextMethod()
+    return(addClass(.data, "soma_adat"))
+  }
+
+  if ( !is_intact_attr(x) || (length(j) == 1L && j > 0 )) {
+    # if 1) attributes already broken OR
+    #    2) extracting a single column
+    #       this behavior may change to match `tbl_df` class
+    #       where `drop = FALSE` by default
+    return(NextMethod(drop = drop))
   }
 
+  # otherwise sub-setting columns; attributes must be considered
```

Comment:
Is this comment in reference to the inserted section above, or the line below? Is the space between this and line 75 intentional?